### ISSUE TITLE
Attempting to attach when already attached doesn't return error code. (SC-573)

### ISSUE THREAD

**Issue by robert-ancell** [canonical/ubuntu-advantage-client#1867]
Attaching with an invalid token returns 1:
```
$ sudo ua attach x
Invalid token. See https://ubuntu.com/advantage
$ echo $?
1
```

Successfully attaching returns 0:
```
$ sudo ua attach <token>
This machine is now attached to '...'

SERVICE       ENTITLED  STATUS    DESCRIPTION
cis           yes       n/a       Center for Internet Security Audit Tools
esm-infra     yes       n/a       UA Infra: Extended Security Maintenance (ESM)
fips          yes       n/a       NIST-certified core packages
fips-updates  yes       n/a       NIST-certified core packages with priority security updates
livepatch     yes       n/a       Canonical Livepatch service
$ echo $?
0
```

Attempting to attach when already attached prints an error, but returns 0:
```
$ sudo ua attach <token>
This machine is already attached to '...'
To use a different subscription first run: sudo ua detach.
$ echo $?
0
```

I would expect the last case to have returned 1 (i.e. an error).

Because the last case returns 0, we don't currently detect this as an error when running the UA client for GUI applications.

**Comment by orndorffgrant** [comment]
Thanks for the feedback @robert-ancell !

I think this was intentional because (in the human driven UX scenario), the user asked to be attached, and they are attached, so it's not a failure per se.

However, I think it would be good to distinguish that situation from a normal successful attach operation like you suggest.

I also don't want to return the same error code as "true" failure though.

What if we return `2` in the case where the machine is already attached? Would that work for you?
(cc @lucasmoura @renanrodrigo)

**Comment by renanrodrigo** [comment]
I understand there is also a reason for it to be 0 on PRO machines in the past, but looking at it right now it makes sense for me to make it `2`.

**Comment by robert-ancell** [comment]
Returning 2 works fine for me, I'm currently concluding an operation succeeded if the return value is 0. I'm not reading the stdout/stderr text, as that will be better resolved in #1812.
